### ISSUE TITLE
v1.3 (@4f988b8)  - First call to Manager.SharedInstance.GetDatabase causes Fatal signal 11 (SIGSEGV)

### ISSUE THREAD

**Issue by Steven-Mark-Ford** [couchbase/couchbase-lite-net#708]
I have done a fresh clone/build of 1.3 (@4f988b8) but crashing with a Fatal signal 11 (SIGSEGV) on first call to Manager.SharedInstance.GetDatabase.

-Visual Studio 2015
-Xamarin Android 6.1.1.1
-Debugging via USB using a Nexus 5
-Project building against and targeting Android 5.1

This is how I replicate the issue:
1. Build Couchbase.Lite.Android (Release) and  @master 4f988b8ece07c6de6893910fc5bb12d8848614ba (2016/07/28)
2. Referenced the dlls in Couchbase.Lite.Android\bin\Release
3. Install the dependencies via package manager i.e:
```
<package id="Newtonsoft.Json" version="7.0.1" targetFramework="MonoAndroid403" />
<package id="SharpZipLib.Portable" version="0.86.0.0003" targetFramework="MonoAndroid403" />
<package id="Stateless" version="2.5.53.0" targetFramework="MonoAndroid403"
```

Similarly for the storage engine (chose e.g. sql lite here):
4. Build storage.systemsqlite.droid (Release)
5. Referenced the dlls in StorageEngines\SystemSQLite\storage.systemsqlite.droid\bin\Release
6.Install the dependencies via package manager i.e:
```
<package id="SQLitePCL.raw" version="0.9.1" targetFramework="monoandroid403" />
<package id="SQLitePCL.ugly" version="0.9.1" targetFramework="monoandroid403" />
```
7.Make a call to Manager.SharedInstance.GetDatabase 

Get the following Exception:
`
07-29 20:48:04.620 E/mono-rt (24708): 
Attempting native Android stacktrace:
07-29 20:48:04.620 E/mono-rt (24708): 

07-29 20:48:04.627 E/mono-rt (24708):  at ???+0 [0x9e9177d8]

07-29 20:48:04.633 E/mono-rt (24708):  at ???+0 [0x0]
07-29 20:48:04.640 E/mono-rt (24708): 
07-29 20:48:04.640 E/mono-rt (24708): =================================================================
07-29 20:48:04.640 E/mono-rt (24708): Got a SIGSEGV while executing native code. This usually indicates
07-29 20:48:04.640 E/mono-rt (24708): a fatal error in the mono runtime or one of the native libraries 
07-29 20:48:04.640 E/mono-rt (24708): used by your application.
07-29 20:48:04.640 E/mono-rt (24708): =================================================================
07-29 20:48:04.640 E/mono-rt (24708): 
07-29 20:48:04.640 F/libc    (24708): Fatal signal 11 (SIGSEGV), code 2, fault addr 0x9aec4fac in tid 24799 (Thread-12730)`
						
Logs from Couchbase:
Tag: DATABASE (Database) Message: No pending doc expirations
Tag: TASK SCHEDULING (SingleTaskThreadpoolScheduler) Message: New scheduler created

**Comment by borrrden** [comment]
Does this happen with the nuget package on the CI server too?

**Comment by Steven-Mark-Ford** [comment]
I will give it a try. I have never installed nuget packages manually outside of the official nuget repo before. Do I just point to the CI server? or do I have to download it manually and run a different command from the package manager console?

**Comment by borrrden** [comment]
It's really easy. See this page under the section create local feed https://docs.nuget.org/create/hosting-your-own-nuget-feeds

**Comment by borrrden** [comment]
Also you said that you are using a Nexus 5, does it happen to be running API 24+?  There is a new rule on Android N that disallows using the system SQLite native library.  I'm going to publish a new library for using a packaged SQLite soon, but in the meantime using SQLCipher is an alternative.

**Comment by Steven-Mark-Ford** [comment]
The nexus 5 is running API 22
I will try the packages on CI server today and provide feedback.

**Comment by borrrden** [comment]
In the meantime 1.3 was released to nuget.org so you might as well get that first ;)

**Comment by Steven-Mark-Ford** [comment]
Yip saw that :) but thanks for the heads up.

**Comment by Steven-Mark-Ford** [comment]
Still got issue when running off CB 1.3 from nuget but managed to figure out the exception was caused by an infinite recursive loop happening in the ILogger I had setup. 

Basically, the logger was serialising couchbase log arguments to json but the issue was that when the argument was a Couchbase database and the database object has a property PersistentCookieStore which logs when the database is not open and passes the database as the argument...causing an infinite loop.

So non-issue here. I just made my ILogger implementation aware of potential recursive calls when serializing log arguments.
